### DISCUSSION TITLE
[Idea] [POC] Introduce environment types

### DISCUSSION THREAD

**Discussion by PululuK** [PrestaShop/PrestaShop#34039] [Ideas]
### The problem

Ability to configure several types of environments to allow developers to adapt the behavior of the store depending on the environmen.

### Shop > Params  :

![Capture d’écran du 2023-09-24 01-02-12](https://github.com/PrestaShop/PrestaShop/assets/16455155/9d34b06c-d9cf-40d6-8d81-4a2f6eaef5e8)

![Capture d’écran du 2023-09-24 01-02-57](https://github.com/PrestaShop/PrestaShop/assets/16455155/5f3a67c2-0167-4d89-a072-f076609203ad)

### Header : Notify the current environment type in the shop header

![Capture d’écran du 2023-09-24 01-05-35](https://github.com/PrestaShop/PrestaShop/assets/16455155/e7005c43-b8b0-4b97-83af-2347640eb588)

![Capture d’écran du 2023-09-24 01-05-13](https://github.com/PrestaShop/PrestaShop/assets/16455155/c56daba0-3d8c-485b-914c-2699ff1c4da8)

# FO

#### smarty : 

- `{$environment.id}`
- `{$environment.name}`
- `{$environment.description}`
- `{$environment.color_code}`
- `{$environment.is_production}`

#### javascript

- `prestashop.environment.id`
- `prestashop.environment.name`
- `prestashop.environment.description`
- `prestashop.environment.color_code`
- `prestashop.environment.is_production`

# Concrete implementations : 

### Robot text generation

Don't index the shop if current confiured environment is not production.

![Capture d’écran du 2023-09-24 01-16-12](https://github.com/PrestaShop/PrestaShop/assets/16455155/3bbedbea-8429-4a03-baca-38c9c53712dc)

### Auther cases : 

- Enable some features acording the current environment type, for example the experimental features.
- etc etc

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by jolelievre** [comment]
That's a notion that is dependent on your infrastructure, I don't think PrestaShop as a CMS should be responsible of this part. I think very few people would actually use it which means it would mostly complexify an already very complicated code for very few added value.

PrestaShop already does way too many things outside of its supposed scope, I'm not in favor of adding more.

**Comment by MatShir** [comment]
I like the idea, I see many applications. But for me, we need to look first to all the conf that are not editable yet ! Make a deep clean and then we could see for a master conf profil.
